compiler/rustc_typeck/src/collect/item_bounds.rs

```diff
@@ -86,7 +86,8 @@ pub(super) fn explicit_item_bounds(
     def_id: DefId,
 ) -> &'_ [(ty::Predicate<'_>, Span)] {
     let hir_id = tcx.hir().local_def_id_to_hir_id(def_id.expect_local());
-    match tcx.hir().get(hir_id) {
+    let node = if let Some(node) = tcx.hir().find(hir_id) { node } else { return &[] };
```

Comment:
I don't like this change too much. Could you rather check at encoding time that we only invoke this for associated types of trait declarations?

I think it would be fine to add a field to the `AssocTy` variant of `DefKind` to encode this information.